src/main/webapp/app/upload/upload.component.ts

```diff
@@ -1,23 +1,101 @@
-import { Component } from '@angular/core';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
 import { Router } from '@angular/router';
-import { UploadEvent, UploadFile, FileSystemFileEntry } from 'ngx-file-drop';
+import { FileSystemFileEntry, UploadEvent, UploadFile } from 'ngx-file-drop';
 import { UploadService } from './upload.service';
+import { Pool } from 'app/pool/pool';
+import { PoolListService } from 'app/pool/pool-list/pool-list.service';
+
+enum PoolType {
+    New,
+    Old
+}
 
 @Component({
     selector: 'jhi-home',
     templateUrl: './upload.component.html',
-    styleUrls: ['upload.css']
+    styleUrls: ['upload.css'],
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class UploadComponent {
     error: string;
     uploadResponse = { status: '', message: '' };
 
+    currentPool: Map<String, Pool> = new Map<String, Pool>();
+    newPool: Map<String, Pool> = new Map<String, Pool>();
+
+    public poolType = PoolType;
+    public isCollapsed = true;
     public files: UploadFile[] = [];
-    file: File;
 
-    constructor(private uploadService: UploadService, private router: Router) {}
+    public currentConflictPool: Pool[] = [];
+    public newConflictPool: Pool[] = [];
+    public merged: Pool[] = [];
+    public choosePattern = new Map();
+    public file: File;
+
+    constructor(
+        private uploadService: UploadService,
+        private router: Router,
+        private poolService: PoolListService,
+        private changeDetectorRef: ChangeDetectorRef
+    ) {}
+
+    async onSubmit() {
+        if (!this.file) {
+            return (this.error = 'Please insert file');
+        }
+        this.clearField();
+        const formData = new FormData();
+        formData.append('file', this.file);
+
+        const oldOnesPromise = this.poolService.getPools().toPromise();
+        const newOnesPromise = this.uploadService.parse(formData).toPromise();
+
+        try {
+            (await oldOnesPromise).body.forEach(pool => this.currentPool.set(pool.poolId, pool));
+            (await newOnesPromise).body.forEach(pool => this.newPool.set(pool.poolId, pool));
+        } catch (exception) {
+            this.uploadService.addAlert('danger', 'upload.fetch-err');
+        }
+        this.getConflicts();
+        this.isCollapsed = false;
```

Comment:
Maybe it would be better to change this if file is valid csv

Reply:
fixed